app/graphql/types/hmis_schema/has_projects.rb

```diff
@@ -0,0 +1,26 @@
+module Types
+  module HmisSchema
+    module HasProjects
+      extend ActiveSupport::Concern
+
+      class_methods do
+        def projects_field(name = :projects, description = nil, without_args: [], **override_options, &block)
+          default_field_options = { type: [Types::HmisSchema::Project], null: false, description: description }
+          field_options = default_field_options.merge(override_options)
+          field(name, **field_options) do
+            argument :exclude_confidential, GraphQL::Types::Boolean, required: false unless without_args.include? :exclude_confidential
+            argument :project_types, [Types::HmisSchema::ProjectType], required: false unless without_args.include? :project_types
+            instance_eval(&block) if block_given?
+          end
+        end
+      end
+
+      def resolve_projects(scope = object.projects, user: current_user, project_types: nil, exclude_confidential: false)
+        projects_scope = scope.viewable_by(user)
+        projects_scope = projects_scope.with_project_type(project_types) if project_types.present?
+        projects_scope = projects_scope.non_confidential if exclude_confidential && !user.can_view_confidential_project_names?
+        projects_scope
+      end
```

Comment:
Same goes for this.  We definitely can't use `Project.viewable_by(user)` in this context, it should be abstracted out into a new class something like `HmisApi::Project` (not sure on the namespace) with its own `viewable_by` scope.  
The main takeaway is that while we need to re-use a bunch of code, we don't want to lean on the existing classes in most cases because the permissions and access are very different for the HMIS.